src/index.ts

```diff
@@ -16,6 +16,43 @@ const rl = createInterface({
   output: process.stdout
 })
 
+const promiseTimeout = function (ms: number, promise: Promise<any>) {
+  const timeout = new Promise((resolve, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id)
+      reject(new Error(`Timed out in ${ms}ms`))
+    }, ms)
+  })
+
+  return Promise.race([
+    promise,
+    timeout
+  ])
+}
+
+function connectToBootnode (node: any, bootNodeAddr: (string | number)) {
+  logger.info(`Trying to connect to ${bootNodeAddr}`)
+  const dialPromise = new Promise<any>((resolve, reject) => {
+    node.dial(new Multiaddr(bootNodeAddr), (err: any, conn: any) => {
+      if (err) {
+        logger.warn(`Error connecting to ${bootNodeAddr}: `, err)
+        reject(err)
+      } else {
+        resolve(conn)
+      }
+    })
+  })
+  return promiseTimeout(300, dialPromise)
+}
+
+function processBootnodes (node: any, bootNodeAddresses: (string | number)[]) {
+  return bootNodeAddresses.reduce((accumulatorPromise: Promise<any>, nextMultiAddress: (string | number)) => {
```

Comment:
Can the `nextMultiAddress` be `number`?

Reply:
linter prank me... it is (string | Buffer | Multiaddr)

Reply:
That is true, but if you look where it comes from (this is the `bootNodeAddresses` arg) it is a string only, right? :)